src/main/scala/io/iohk/ethereum/network/rlpx/MessageCodec.scala

```diff
@@ -22,8 +24,12 @@ class MessageCodec(frameCodec: FrameCodec, messageDecoder: MessageDecoder, proto
   // MessageCodec is only used from actor context so it can be var
   @volatile
   private var remotePeerP2pVersion: Option[Long] = None
+  @volatile
+  private var remotePeerProtocolVersion: Option[Version] = None
+
+  private val temporalDecoder = messageDecoder.fromBytes(_, _, protocolNegotiator.protocolVersion)
```

Comment:
```suggestion
  private val negotiatorProtocolVersionFromBytes = messageDecoder.fromBytes(_, _, protocolNegotiator.protocolVersion)
```

or

why not just remove the private val and use that directly in the code?